src/topology/schema.rs

```diff
@@ -25,6 +27,21 @@ use crate::{config::OutputId, topology};
 /// Finally, The merged definition (named `Definition 1 & 2`), and `Definition 4` are merged
 /// together to produce the new `Definition` returned by this method.
 pub(super) fn merged_definition(inputs: &[OutputId], config: &topology::Config) -> Definition {
+    let mut cache = HashMap::default();
+
+    inner_merged_definition(inputs, config, &mut cache)
+}
+
+fn inner_merged_definition(
+    inputs: &[OutputId],
+    config: &topology::Config,
+    cache: &mut HashMap<Vec<OutputId>, Definition>,
```

Comment:
Does the order of the `OutputId` matter? If not, would we increase cache hits by sorting this first?